src/components/destination_list.js

```diff
@@ -17,9 +18,11 @@ const DestinationList = ({
   height,
   unselectItems,
   selectedItemRenderer,
-  noItemsRenderer
+  noItemsRenderer,
+  withGrouping
 }) => {
   const SelectionStatusRenderer = selectionStatusRenderer;
+  const updatedSelectedItems = buildItemGrouping(selectedItems, withGrouping);
```

Comment:
as this name contains `grouping` part, I would move withGrouping condition check from it to here
`const updatedSelectedItems = withGrouping ? buildItemGrouping(selectedItems) : selectedItems;`